src/tensorflow.cc

```diff
@@ -2024,88 +2023,98 @@ ModelInstanceState::ProcessRequests(
     for (const auto& name : model_output_names) {
       TRITONTF_Tensor* output_tensor = output_tensor_itr->tensor_;
 
-      const BatchOutput* batch_output = StateForModel()->FindBatchOutput(name);
-      if (batch_output == nullptr) {
-        TRITONTF_DataType tf_datatype = TRITONTF_TensorDataType(output_tensor);
-        TRITONTF_Shape* tf_shape = TRITONTF_TensorShape(output_tensor);
-
-        const TRITONSERVER_DataType datatype = ConvertDataType(tf_datatype);
-
-        // batchn_shape holds the shape of the entire tensor batch, but
-        // is overwritten below and used as the shape for each response
-        // output.
-        std::vector<int64_t> batchn_shape;
-        batchn_shape.reserve(tf_shape->rank_);
-        for (size_t itr = 0; itr < tf_shape->rank_; itr++) {
-          const int64_t dim = tf_shape->dims_[itr];
-          batchn_shape.push_back(dim);
-        }
+      // For certain input data, the model can return an empty tensor
+      // as output. Make sure the output is available before trying
+      // to copy it out.
+      if (TRITONTF_TensorData(output_tensor) != nullptr) {
```

Comment:
I think this whole block would read easier if instead of checking the negative we just did something like this instead if logically equivalent:
```cpp
      if (TRITONTF_TensorData(output_tensor) == nullptr) {
        continue
      }
      
      ... // rest of the code unchanged
```

Reply:
I had a bug in my previous fix. I need to still advance the iterator. I can still have the following if you'd like:
```
if (TRITONTF_TensorData(output_tensor) == nullptr) {
        output_tensor_itr = output_tensor_itr->next_;
        continue
      }
      
      ... // rest of the code unchanged
```

However, we must avoid duplicating code if possible -- in this case advancing the iterator.

Reply:
If going with this, we could also advance the iterator right after getting the tensor at start of loop to avoid duplication:
```cpp
    for (const auto& name : model_output_names) {
      TRITONTF_Tensor* output_tensor = output_tensor_itr->tensor_;
      // Advance iterator here instead of at end (if valid) to avoid code duplication
      output_tensor_itr = output_tensor_itr->next_;
      
      if (TRITONTF_TensorData(output_tensor) == nullptr) { 
        continue 
      } 
      
      ... // rest of the code unchanged
    }
```

But if you prefer to leave the iterator advancement at the end and leave your current code as is checking (!= nulltpr), that's fine